common/types/admin.go

```diff
@@ -364,3 +364,54 @@ func (v *ListDynamicConfigRequest) SerializeForLogging() (string, error) {
 type ListDynamicConfigResponse struct {
 	Entries []*DynamicConfigEntry `json:"entries,omitempty"`
 }
+
+type IsolationGroupState int
+
+const (
+	IsolationGroupStateInvalid IsolationGroupState = iota
+	IsolationGroupStateHealthy
+	IsolationGroupStateDrained
+)
+
+type IsolationGroupPartition struct {
+	Name  string
+	State IsolationGroupState
+}
+
+// IsolationGroupConfiguration is an internal representation of a set of
+// isolation-groups as a mapping and may refer to either globally or per-domain (or both) configurations.
+// and their statuses. It's redundantly indexed by IsolationGroup name to simplify lookups.
+//
+// For example: This might be a global configuration persisted
+// in the config store and look like this:
+//
+//	IsolationGroupConfiguration{
+//	  "isolationGroup1234": {Name: "isolationGroup1234", Status: IsolationGroupStatusDrained},
+//	}
+//
+// Indicating that task processing isn't to occur within this isolationGroup anymore, but all others are ok.
+type IsolationGroupConfiguration map[string]IsolationGroupPartition
```

Comment:
it's quite different from idl definition

```
struct IsolationGroupConfiguration {
  10: optional list<IsolationGroupPartition> isolationGroups
}
```

Reply:
yes, this was intentional: Most of the checking of values in the isolation groups are set membership operations (is this zone drained) and representing them as a list would have meant iterating over them multiple times. Not a huge cost, but also not really worthwhile. 

However, representing them as a map by name introduces a redundant name, which I didn't want to expose in IDL, thus I was going to transform it here in the mapping, with the internal implementation better suited for membership checks, and the public IDL less confusing to understand